lib/command.h

```diff
@@ -399,6 +402,8 @@ extern void cmd_exit(struct vty *vty);
 extern int cmd_list_cmds(struct vty *vty, int do_permute);
 
 extern int cmd_hostname_set(const char *hostname);
+extern const char *hostname_get(void);
+extern const char *domainname_get(void);
```

Comment:
Nit: please rename these to `cmd_hostname_get` & `cmd_domainname_get` to fit the API naming, and add a `cmd_domainname_set`. (The `host` variable is rarely accessed directly from outside lib/, I'd hope we can cut that down to zero at some point and make it private.)

Additionally, we should add a comment here that hostname will never return NULL, but domainname can.

Reply:
(now that i look at it... hostname_get *can* return NULL?)

Reply:
yes it can return NULL in case the hostname is not set in either /etc/hostname or via config in frr.

Reply:
changed hostname_get to cmd_hostname_get; changed domainname_get to cmd_domainname_get